From b088319bfd7743ecf2b19517fb577b0dd0590a43 Mon Sep 17 00:00:00 2001
From: tchew6 <tong.liang.chew@intel.com>
Date: Thu, 24 Jun 2021 13:20:53 +0800
Subject: [PATCH 07/11] iris: Included resource scannout as pre-condition
 checking

The renderonly_scanout object exits for two reasons:
 	 - Do any special treatment for a scanout resource like importing the GPU
	   resource into the scanout hw.
	 - Make it easier for a gallium driver to detect if anything special needs
	   to be done in flush_resource(..) like a resolve to linear

For SRIOV cases, we are not allowed resource scannout to be NULL before
it is getting handled

Upstream URL: NA
Upstream Status: Innapropriate

OSV: https://jira.devtools.intel.com/browse/PKT-19619

Signed-off-by: Chew, Tong Liang <tong.liang.chew@intel.com>
Signed-off-by: Mazlan, Hazwan Arif <hazwan.arif.mazlan@intel.com>
---
 src/gallium/drivers/iris/iris_resource.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/gallium/drivers/iris/iris_resource.c b/src/gallium/drivers/iris/iris_resource.c
index 37c7eacc75ac..78d26c57375d 100644
--- a/src/gallium/drivers/iris/iris_resource.c
+++ b/src/gallium/drivers/iris/iris_resource.c
@@ -1148,12 +1148,12 @@ iris_resource_create_for_image(struct pipe_screen *pscreen,
 {
    struct iris_screen *screen = (struct iris_screen *)pscreen;
    const struct intel_device_info *devinfo = screen->devinfo;
+   struct pipe_resource *pres;
 
    if (screen->ro &&
        (templ->bind & (PIPE_BIND_DISPLAY_TARGET |
                        PIPE_BIND_SCANOUT | PIPE_BIND_SHARED))) {
-      struct pipe_resource *pres =
-             iris_resource_create_renderonly(pscreen, templ);
+      pres = iris_resource_create_renderonly(pscreen, templ);
       if (pres)
          return pres;
    }
@@ -1965,7 +1965,7 @@ iris_resource_get_handle(struct pipe_screen *pscreen,
    case WINSYS_HANDLE_TYPE_KMS: {
       iris_gem_set_tiling(bo, &res->surf);
 
-      if (screen->ro) {
+      if (screen->ro && res->scanout) {
          return renderonly_get_handle(res->scanout, whandle);
       }
       /* Because we share the same drm file across multiple iris_screen, when
-- 
2.34.1

